homeassistant/components/trafikverket_train/sensor.py

```diff
@@ -153,6 +178,23 @@ def __init__(
         self._to_station = to_station
         self._weekday = weekday
         self._time = departuretime
+        self._attr_device_info = DeviceInfo(
+            entry_type=DeviceEntryType.SERVICE,
+            identifiers={(DOMAIN, entry_id)},
+            manufacturer="Trafikverket",
+            model="v1.2",
+            name=name,
+            configuration_url="https://api.trafikinfo.trafikverket.se/",
+        )
+
+    @property
+    def unique_id(self) -> str:
+        """Return unique id."""
+        timestr = "" if self._time is None else str(self._time)
+        return (
+            f"{self._from_station.casefold().replace(' ', '')}-{self._to_station.casefold().replace(' ', '')}"
+            f"-{timestr.casefold().replace(' ', '')}-{self._weekday}"
+        )
```

Comment:
This is repeated quite a few times in the integration.  It would be good to make it a function so we don't risk a refactoring error in the future